core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -189,7 +189,7 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
     val filteredResources =
       remainingResources.filter(r => r.getType != Value.Type.SCALAR || r.getScalar.getValue > 0.0)
 
-    (filteredResources.toList, requestedResources.toList)
+    (filteredResources.toList.asJava, requestedResources.toList.asJava)
```

Comment:
You don't need `toList.asJava`. `asJava` is enough, and saves a round of copying.